File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int start = 0, end = nums.length - 1, ans = 0;
        while(start <= end) {
            int mid = start + (end - start) / 2;
            if(nums[mid] == target) return mid;
            else if(nums[mid] < target) {
                ans = mid + 1;
                start = mid + 1;
            }
            else end = mid - 1;
        }
        return ans;
    }
}

File: 35-search-insert-position/35-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        start, end, pos = 0, len(nums) - 1, 0
        while start <= end:
            mid = start + (end - start) // 2
            if nums[mid] == target: 
                return mid
            elif nums[mid] < target:
                pos = mid + 1
                start = mid + 1
            else:
                end = mid - 1
        return pos